app/src/main/java/com/keylesspalace/tusky/util/FlowExtensions.kt

```diff
@@ -63,3 +73,45 @@ fun <T> Flow<T>.throttleFirst(timeout: Duration, timeSource: TimeSource = TimeSo
             }
         }
     }
+
+fun <T> Flow<T>.observe(
+    scope: CoroutineScope,
+    collector: FlowCollector<T>
+): Job = scope.launch {
+    collect(collector)
+}
+
+fun <T> Flow<T>.observeLatest(
+    scope: CoroutineScope,
+    action: suspend (value: T) -> Unit
+): Job = scope.launch {
+    collectLatest(action)
+}
+
+fun <T> Flow<T>.observe(
+    lifecycleOwner: LifecycleOwner,
+    collector: FlowCollector<T>? = null
+): Job = lifecycleOwner.lifecycleScope.launch {
+    lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+        if (collector == null) {
+            collect()
+        } else {
+            collect(collector)
+        }
+    }
+}
+
+fun <T> Flow<T>.observeLatest(
+    lifecycleOwner: LifecycleOwner,
+    action: suspend (value: T) -> Unit
+): Job = observeLatest(lifecycleOwner.lifecycleScope, action)
+
+context(LifecycleOwner)
+fun <T> Flow<T>.observe(
+    collector: FlowCollector<T>? = null
+): Job = observe(this@LifecycleOwner, collector)
```

Comment:
If we can use `Fragment`'s `viewLifecycleOwner` instead of its self for all invocations in Fragments?

Reply:
Finished in 38bdb76fd8bfb4981514c0028f9fb693cab3bfc5, feel free to review.